### DISCUSSION TITLE
Implement WebView as a Theia plugin

### DISCUSSION THREAD

**Discussion by Nichole0** [eclipse-theia/theia#14090] [Q&A] [answered]
I have noticed that a web view plugin extension is offered in Theia under the following path: theia/packages/plugin-ext/src/main/browser/webview/webview.ts and I wanted to know how to implement this as a Theia plugin?

 I have tried to use vscode plugin, but found that vscode webview plugin is not compatible with Theia.For example, the Netron plugin for VS Code, which can be found at https://github.com/chingweihsu0809/NetronInVSCode.git, does not work properly in Theia.The widget shown as
![image](https://github.com/user-attachments/assets/6d759657-7023-420a-b7c4-7fbffeec3dd6)

**Comment by msujew** [comment]
Hey @Nichole0,

the webview API is definitely supported in Theia and is supposed to work as expected. I've just tested this specific extension in Theia:

![image](https://github.com/user-attachments/assets/98511a34-f869-488d-88d9-4b34a0a5c38f)

**Comment by Nichole0** [comment]
I’m still having trouble getting the plugin to work. 
Could you please let me know if you had to do any extra setup or configuration to get the plugin working in Theia?
Thanks!

![image](https://github.com/user-attachments/assets/58062d23-16c1-4792-898c-ec28a08740c5)

**Reply by msujew** [reply, marked as answer]
I simply downloaded the extension from the [vscode marketplace](https://marketplace.visualstudio.com/items?itemName=chingweihsu0809.netron-vscode) and installed it into Theia. Additionally, I needed to run `pip install netron`. Aside from that, everything worked out of the box.
